### ISSUE TITLE
Sub-MenuItem does not close when clicked while using Items binding

### ISSUE THREAD

**Issue by wieslawsoltes** [AvaloniaUI/Avalonia#489]
Code to reproduce:

``` XAML
<MenuItem Header="_File" IsVisible="{Binding Project, Converter={Static converters:IsNullConverter.Instance}}">
    <MenuItem Header="_New" Command="{Static Editor:Commands.NewCommand}" CommandParameter="{Binding}"/>
    <Separator/>
    <MenuItem Header="_Open..." Command="{Static Editor:Commands.OpenCommand}"/>
    <Separator/>
    <MenuItem Header="_Recent" Items="{Binding RecentProjects}">
        <MenuItem.DataTemplates>
            <DataTemplate DataType="Recent:RecentFile">
                <MenuItem Header="{Binding Name}" Command="{Static Editor:Commands.OpenCommand}" CommandParameter="{Binding Path}"/>
            </DataTemplate>
        </MenuItem.DataTemplates>
    </MenuItem>
    <Separator/>
    <MenuItem Header="E_xit" Command="{Static Editor:Commands.ExitCommand}"/>
</MenuItem>
```

Screenshot before clicking Sub-MenuItem:

![menuitemissuebefore](https://cloud.githubusercontent.com/assets/2297442/14065592/94b1c98e-f430-11e5-89fe-03f180dc5479.png)

Screenshot after clicking Sub-MenuItem:

![menuitemissue](https://cloud.githubusercontent.com/assets/2297442/14065578/1d3a32d8-f430-11e5-9774-ce6f871d1301.png)

**Comment by wieslawsoltes** [comment]
I have found that this only happens when the bound `Items` collection has changed after used clicked.

```
<MenuItem Header="_Recent" Items="{Binding RecentProjects}">
```

In my case when user clicks on one of the items I open the bound item (a path to a project file) and update accordingly the `RecentProjects` collection so the opened project is first item in the list. As the `RecentProjects` collection is immutable array, so to propagate collection changes I set `RecentProjects` to a new value.
